OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/internal/permissions/impl/NotificationPermissionController.kt

```diff
@@ -64,14 +70,35 @@ internal class NotificationPermissionController(
             )!!
 
     init {
+        this.enabled = notificationsEnabled()
         _requestPermission.registerAsCallback(PERMISSION_TYPE, this)
+        coroutineScope.launch {
+            pollForPermission()
+        }
+    }
+
+    private suspend fun pollForPermission() {
+        while (true) {
+            val enabled = this.notificationsEnabled()
+            if (this.enabled != enabled) { // If the permission has changed without prompting through OneSignal
+                this.enabled = enabled
+                events.fire { it.onNotificationPermissionChanged(enabled) }
+            }
+            delay(1_000) // should be a configurable value for unit tests
```

Comment:
Note, might want to use a waiter here with a timeout to account for waking when the app is foregrounded, and changing the wait time to something very large when the app is backgrounded (to effectively pause the polling).

Reply:
I updated this to use a waiter. The in focus interval is 1 second and the out of focus is currently 30 minutes. I am not sure what a reasonable value is for the out of focus polling. Do we even want to continue polling while in the background?